embedded-graphics/src/style/text_style.rs

```diff
@@ -37,17 +44,172 @@ where
     }
 }
 
+/// Text style builder.
+///
+/// Use this builder to create styles for [`Font`]s. For convenience, the [`text_style!`] macro is
+/// also provided.
+///
+/// # Examples
+///
+/// ## Render some text with a blue background and yellow text color
+///
+/// This uses the [`Font6x8`] font, but [other fonts] can also be used.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     fonts::{Font6x8, Text},
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     style::{TextStyle, TextStyleBuilder},
+/// };
+///
+/// let style: TextStyle<Rgb565, Font6x8> = TextStyleBuilder::new(Font6x8)
+///     .text_color(Rgb565::YELLOW)
+///     .background_color(Rgb565::BLUE)
+///     .build();
+///
+/// let text = Text::new("Hello Rust!", Point::new(0, 0)).into_styled(style);
+/// ```
+///
+/// ## Render black text on white background using macros
+///
+/// This uses the [`Font8x16`] font with the [`egtext!`] and [`text_style!`] macros for shorter
+/// code.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     egtext,
+///     fonts::{Font8x16, Text},
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     style::TextStyle,
+///     text_style,
+/// };
+///
+/// let style = text_style!(
+///     font = Font8x16,
+///     text_color = Rgb565::WHITE,
+///     background_color = Rgb565::BLACK
+/// );
+///
+/// let text = Text::new("Hello Rust!", Point::new(0, 0)).into_styled(style);
+/// ```
+///
+/// ## Transparent background
+///
+/// If a property is ommitted, it will default to `None` in the resulting `TextStyle` returned by
```

Comment:
Not all values might default to `None` in the future. I would just mention that they stay at the default value.